android/clientlib/src/main/java/com/solana/mobilewalletadapter/clientlib/protocol/MobileWalletAdapterClient.java

```diff
@@ -331,6 +335,14 @@ protected AuthorizationResult processResult(@Nullable Object o)
                         walletUriBaseStr + "'; expected a 'https' URI");
             }
 
+            final Uri walletIcon;
+            try {
+                final String walletIconStr = jo.getString(ProtocolContract.RESULT_WALLET_ICON);
+                walletIcon = Uri.parse(walletIconStr);
+            } catch (JSONException e) {
+                throw new JsonRpc20InvalidResponseException("expected a wallet_icon");
```

Comment:
Will we run into this case if the wallet endpoint is on an older version, without the DefaultWalletIconProvider changes?

Reply:
oh yes good call